server/server.go

```diff
@@ -2566,85 +2530,62 @@ func readInputCompilePostV1(r io.ReadCloser) (*compileRequest, *types.ErrorV1) {
 	return result, nil
 }
 
-func renderBanner(w http.ResponseWriter) {
-	fmt.Fprintln(w, `<pre>
+var indexHTML, _ = template.New("index").Parse(`
+<html>
+<head>
+<script type="text/javascript">
+function query() {
+	params = {
+		'query': document.getElementById("query").value,
+	}
+	if (document.getElementById("input").value !== "") {
+		try {
+			params["input"] = JSON.parse(document.getElementById("input").value);
+		} catch (e) {
+			document.getElementById("result").innerHTML = e;
+			return;
+		}
+	}
+	body = JSON.stringify(params);
+	opts = {
+		'method': 'POST',
+		'body': body,
+	}
+	fetch(new Request('/v1/query', opts))
+		.then(resp => resp.json())
+		.then(json => {
+			str = JSON.stringify(json, null, 2);
+			document.getElementById("result").innerHTML = str;
+		});
+}
+</script>
+</head>
+</body>
+<pre>
  ________      ________    ________
 |\   __  \    |\   __  \  |\   __  \
 \ \  \|\  \   \ \  \|\  \ \ \  \|\  \
  \ \  \\\  \   \ \   ____\ \ \   __  \
   \ \  \\\  \   \ \  \___|  \ \  \ \  \
    \ \_______\   \ \__\      \ \__\ \__\
     \|_______|    \|__|       \|__|\|__|
-	</pre>`)
-	fmt.Fprintln(w, "Open Policy Agent - An open source project to policy-enable your service.<br>")
-	fmt.Fprintln(w, "<br>")
-}
-
-func renderFooter(w http.ResponseWriter) {
-	fmt.Fprintln(w, "</body>")
-	fmt.Fprintln(w, "</html>")
-}
-
-func renderHeader(w http.ResponseWriter) {
-	fmt.Fprintln(w, "<html>")
-	fmt.Fprintln(w, "<body>")
-}
-
-func renderQueryForm(w http.ResponseWriter, qStrs []string, inputStrs []string, explain types.ExplainModeV1) {
-
-	query := ""
-
-	if len(qStrs) > 0 {
-		query = qStrs[len(qStrs)-1]
-	}
-
-	input := ""
-	if len(inputStrs) > 0 {
-		input = inputStrs[len(inputStrs)-1]
-	}
-
-	explainRadioCheck := []string{"", "", ""}
-	switch explain {
-	case types.ExplainOffV1:
-		explainRadioCheck[0] = "checked"
-	case types.ExplainFullV1:
-		explainRadioCheck[1] = "checked"
-	}
-
-	fmt.Fprintf(w, `
-	<form>
-  	Query:<br>
-	<textarea rows="10" cols="50" name="q">%s</textarea><br>
-	<br>Input Data (JSON):<br>
-	<textarea rows="10" cols="50" name="input">%s</textarea><br>
-	<br><input type="submit" value="Submit"> Explain:
-	<input type="radio" name="explain" value="off" %v>Off
-	<input type="radio" name="explain" value="full" %v>Full
```

Comment:
These radio buttons are no longer there in the new version. Was that a deliberate choice or did you just forget about them?

Reply:
Yeah, mentioned this in one of the commit messages. We can add it back at some point if we want, but it would be nice to think a bit more holistically about it.